db/document.go

```diff
@@ -194,17 +196,30 @@ func (doc *Document) MarshalBodyAndSync() (retBytes []byte, err error) {
 	}
 }
 
-func (doc *Document) MarshalBodyForWebhook() (retBytes []byte, err error) {
+func (doc *Document) IsDeleted() bool {
+	return doc.hasFlag(channels.Deleted)
+}
+
+func (doc *Document) BodyWithSpecialProperties() ([]byte, error) {
 	bodyBytes, err := doc.BodyBytes()
 	if err != nil {
 		return nil, err
 	}
 
-	bodyBytes, err = base.InjectJSONProperties(
-		bodyBytes,
-		base.KVPair{Key: BodyId, Val: doc.ID},
-		base.KVPair{Key: BodyRev, Val: doc.CurrentRev},
-	)
+	kvPairs := []base.KVPair{
+		{Key: BodyId, Val: doc.ID},
+		{Key: BodyRev, Val: doc.CurrentRev},
+	}
+
+	if doc.IsDeleted() {
+		kvPairs = append(kvPairs, base.KVPair{Key: BodyDeleted, Val: true})
+	}
+
+	if doc.hasFlag(channels.Removed) {
```

Comment:
I don't think a document should have channels.Removed in its flags - I think that flag should only be used for change entries.

Reply:
You're right - I think I added this just because I saw the constant, not thinking too much about where it would appear.